rest-service/manager_rest/rest/resources_v3_1/cluster_status.py

```diff
@@ -0,0 +1,100 @@
+#########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+#
+
+import os
+import json
+from flask import current_app
+from datetime import datetime
+
+from manager_rest import manager_exceptions
+from manager_rest.security.authorization import authorize
+from manager_rest.storage import models, get_storage_manager
+from manager_rest.security import SecuredResourceReadonlyMode
+from manager_rest.rest.rest_utils import (get_json_and_verify_params,
+                                          parse_datetime_string)
+
+STATUSES_PATH = '/opt/manager/resources/cluster_status'
+TIMESTAMP_FORMAT = 'please provide valid date. \nExpected format: ' \
+                   'YYYYMMDDHHMM+HHMM or YYYYMMDDHHMM-HHMM i.e: ' \
+                   '201801012230-0500 (Jan-01-18 10:30pm EST)'
+
+
+class ClusterStatus(SecuredResourceReadonlyMode):
+    @staticmethod
+    def _get_request_dict():
+        request_dict = get_json_and_verify_params({
+            'reporting_freq': {'type': int},
+            'report': {'type': dict},
+            'timestamp': {'type': basestring}
+        })
+        return request_dict
+
+    @staticmethod
+    def _verify_report_newer_than_current(report_time, path):
+        with open(path) as current_report_file:
+            current_report = json.load(current_report_file)
+        if report_time < parse_datetime_string(current_report['timestamp']):
+            current_app.logger.error('The new report timestamp {0} is before'
+                                     ' the current report timestamp, {1}'.
+                                     format(report_time, TIMESTAMP_FORMAT))
```

Comment:
TIMESTAMP_FORMAT is not really useful for anyone, we are just interested in the beginning of the sentence